sources/packages/backend/apps/api/src/services/application-bulk-withdrawal/application-bulk-withdrawal-import.service.ts

```diff
@@ -0,0 +1,114 @@
+
+type ApplicationDataMap = Record<string, ApplicationData>;
+
+/**
+ * Handles the application withdrawal bulk insert preparation.
+ */
+@Injectable()
+export class ApplicationBulkWithdrawalImportService {
+  constructor(
+    @InjectRepository(Application)
+    private readonly applicationRepo: Repository<Application>,
+  ) {}
+
+  /**
+   * Generates the application bulk withdrawal validation model.
+   * @param textValidations text validation models to be converted to the validation models.
+   * @param textHeader text header model.
+   * @param institutionId institution id.
+   * @returns application bulk withdrawal models to be validated and persisted.
+   */
+  async generateApplicationBulkWithdrawalValidationModels(
+    textValidations: ApplicationWithdrawalTextValidationResult[],
+    textHeader: ApplicationBulkWithdrawalHeader,
+    institutionId: number,
+  ): Promise<ApplicationBulkWithdrawalValidationModel[]> {
+    const applicationDataMap: ApplicationDataMap = {};
+    const applicationNumbers = textValidations.map(
+      (textValidation) => textValidation.textModel.applicationNumber,
+    );
+    const applicationValidationData = await this.getApplicationValidationData(
+      applicationNumbers,
+      institutionId,
+    );
+    applicationValidationData.forEach((record) => {
+      applicationDataMap[record.applicationNumber] = {
+        applicationStatus: record.applicationStatus,
+        isArchived: record.isArchived,
+        sin: record.student.sinValidation.sin,
+        locationId: record.location.id,
+        locationCode: record.location.institutionCode,
+        hasPreviouslyBeenWithdrawn: !!record.studentScholasticStandings.length,
+      };
+    });
+    const validationModels = textValidations.map((textValidation) => {
+      const validationModel = {} as ApplicationBulkWithdrawalValidationModel;
+      const applicationData =
+        applicationDataMap[textValidation.textModel.applicationNumber];
+      if (applicationData) {
+        validationModel.applicationFound = true;
+        validationModel.studentSINMatch =
+          textValidation.textModel.sin === applicationData.sin;
+        validationModel.hasCorrectInstitutionCode =
+          textHeader.originator === applicationData.locationCode;
+        validationModel.isArchived = applicationData.isArchived;
+        validationModel.applicationStatus = applicationData.applicationStatus;
+        validationModel.hasPreviouslyBeenWithdrawn =
+          applicationData.hasPreviouslyBeenWithdrawn;
+        validationModel.isRecordMatch = validationModel.studentSINMatch;
+      } else {
+        validationModel.applicationFound = false;
+      }
+      validationModel.sin = textValidation.textModel.sin;
+      validationModel.applicationNumber =
+        textValidation.textModel.applicationNumber;
+      validationModel.withdrawalDate = textValidation.textModel.withdrawalDate;
+      return validationModel;
+    });
+    return validationModels;
```

Comment:
Minor it can be returned directly without using the `validationModels` variable.